File: app/src/androidTest/java/com/an/github/db/GithubDbTest.java
package com.an.github.db;

import android.support.test.runner.AndroidJUnit4;

import com.an.github.data.local.entity.GithubEntity;
import com.an.github.util.MockTestUtil;

import org.junit.Assert;
import org.junit.Test;
import org.junit.runner.RunWith;

import java.util.List;

@RunWith(AndroidJUnit4.class)
public class GithubDbTest extends DbTest {

    @Test
    public void insertAndReadPostsTest() {
        List<GithubEntity> repositories = MockTestUtil.mockRepositories();
        db.githubDao().insertRepositories(repositories);

        List<GithubEntity> storedPosts1 = db.githubDao().getRepositoriesByPage(1l);
        Assert.assertEquals(2l, storedPosts1.size());
        Assert.assertEquals(new Long(1), storedPosts1.get(0).getId());

        List<GithubEntity> storedPosts2 = db.githubDao().getRepositoriesByPage(2l);
        Assert.assertEquals(1l, storedPosts2.size());
        Assert.assertEquals(new Long(3), storedPosts2.get(0).getId());
    }


    @Test
    public void emptyPostsTest() {
        List<GithubEntity> repositories = MockTestUtil.mockRepositories();
        db.githubDao().insertRepositories(repositories);

        List<GithubEntity> storedPosts = db.githubDao().getRepositoriesByPage(3l);
        Assert.assertEquals(0, storedPosts.size());
    }
}

File: app/src/androidTest/java/com/an/github/db/GithubDbTest.kt
package com.an.github.db
import com.an.github.util.MockTestUtil

import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith

//@RunWith(AndroidJUnit4::class)
class GithubDbTest : DbTest() {

//    @Test
//    fun insertAndReadPostsTest() {
//        val repositories = MockTestUtil.mockRepositories()
//        db.githubDao().insertRepositories(repositories)
//
//        val storedPosts1 = db.githubDao().getRepositoriesByPage(1L)
//        Assert.assertEquals(2L, storedPosts1.size.toLong())
//        Assert.assertEquals(1, storedPosts1[0].id)
//
//        val storedPosts2 = db.githubDao().getRepositoriesByPage(2L)
//        Assert.assertEquals(1L, storedPosts2.size.toLong())
//        Assert.assertEquals(3, storedPosts2[0].id)
//    }
//
//
//    @Test
//    fun emptyPostsTest() {
//        val repositories = MockTestUtil.mockRepositories()
//        db.githubDao().insertRepositories(repositories)
//
//        val storedPosts = db.githubDao().getRepositoriesByPage(3L)
//        Assert.assertEquals(0, storedPosts.size.toLong())
//    }
}
